src/util/usertools.c

```diff
@@ -572,10 +573,23 @@ sss_fqname(char *str, size_t size, struct sss_names_ctx *nctx,
 
 errno_t sss_user_by_name_or_uid(const char *input, uid_t *_uid, gid_t *_gid)
 {
+    static struct sss_nss_ops nss_ops;
     uid_t uid;
     errno_t ret;
     char *endptr;
-    struct passwd *pwd;
+    struct passwd pwd = { 0 };
+    int errnop = 0;
+    enum nss_status status;
+    char s_nss_buffer[NSS_BUFFER_SIZE];
+
+    if (!nss_ops.dl_handle) {
+        ret = sss_load_nss_pw_symbols(&nss_ops);
+        if (ret != EOK) {
+            DEBUG(SSSDBG_FATAL_FAILURE, "Unable to load NSS symbols [%d]: %s\n",
```

Comment:
`SSSDBG_OP_FAILURE`
This is just a helper function, severity of it's fail depends on usage context.

Reply:
It makes sense to clean up `nss_ops.dl_handle` here: `sss_load_nss_symbols()` can fail but set `dl_handle`.
This is probably bad design of `sss_load_nss_symbols()` :) but you need to cope with it.

Reply:
I prefer to close it inside `sss_load_nss_symbols()` .. Please take a look at the new version of the changes.